lms/djangoapps/instructor/tests/test_enrollment.py

```diff
@@ -576,33 +574,35 @@ def test_bad_action(self):
 
 
 @attr('shard_1')
-class TestGetEmailParamsCCX(ModuleStoreTestCase):
+class TestGetEmailParamsCCX(SharedModuleStoreTestCase):
     """
     Test what URLs the function get_email_params for CCX student enrollment.
     """
 
     MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
 
+    @classmethod
     @patch.dict('django.conf.settings.FEATURES', {'CUSTOM_COURSES_EDX': True})
-    def setUp(self):
-        super(TestGetEmailParamsCCX, self).setUp()
+    def setUpClass(cls):
+        super(TestGetEmailParamsCCX, cls).setUpClass()
+        cls.course = CourseFactory.create()
+        cls.coach = AdminFactory.create()
+        role = CourseCcxCoachRole(cls.course.id)
+        role.add_users(cls.coach)
+        cls.ccx = CcxFactory(course_id=cls.course.id, coach=cls.coach)
+        cls.course_key = CCXLocator.from_course_locator(cls.course.id, cls.ccx.id)
```

Comment:
`setUpClass()` should only be used to initialize modulestore backed data (i.e. course objects, their attributes, and their children). All Django ORM model backed data (like `CcxFactory`, users, etc.) *must* be put in the `setUp()` or it will not get properly cleaned up. This is probably why `lms/djangoapps/instructor/tests/test_legacy_raw_download_csv.py` is breaking. You might want to try to run all `lms/djangoapps/instructor/tests/` together at once, instead of just `test_legacy_raw_download_csv.py` in isolation.

What's happening is that since the two users being created here and in the other `setUpClass` below aren't rolled back, the sequence IDs for users is never reset. So when `test_legacy_raw_download_csv.py` creates users, they have user IDs of 3 and 4 instead of 1 and 2. As part of its CSV download test, it looks for output with those specific user IDs. Now that's bad behavior on the part of `test_legacy_raw_download_csv.py` -- it has references to what users it has created and it should be building the expected test string based on those IDs instead of hardcoding the assumption of 1 and 2. That being said, any side-effects leaking out in the test classes is to be avoided. Please move the Django model backed data into `setUp()`.

FWIW, there's a [note](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/tests/django_utils.py#L250) about this in the docstring for SharedModuleStoreTestCase. We should definitely document this better though.